eng/pipelines/templates/jobs/ci.tests.yml

```
@@ -125,6 +125,14 @@ jobs:
           )
         displayName: Set Xcode version
 
+      - pwsh: /usr/bin/xcodebuild -version
+        condition: >-
+          and(
+            succeeded(),
+            contains(variables['OSVmImage'], 'macOS')
```

This variable will give you stronger answers on the OS in use... bonus points if you also fix up the previous step that runs `xcode-select` 
```suggestion 
            eq(variables['Agent.OS'], 'Darwin')
```